drivers/spi/spi_ll_stm32.c

```diff
@@ -122,31 +122,34 @@ static uint32_t bits2bytes(uint32_t bits)
 static __aligned(32) uint32_t dummy_rx_tx_buffer __nocache;
 
 /* This function is executed in the interrupt context */
-static void dma_callback(const struct device *dev, void *arg,
+static void dma_callback(const struct device *dma_dev, void *arg,
 			 uint32_t channel, int status)
 {
-	/* arg directly holds the spi device */
-	struct spi_stm32_data *data = arg;
+	ARG_UNUSED(dma_dev);
+
+	/* arg holds DMA rx/tx channel data
+	 * Passed in spi_stm32_dma_tx/rx_load()
+	 */
+	struct spi_stm32_data *dma_channel_data = arg;
```

Comment:
Would it make any confusion naming a variable _dma_channel_data_  of type  _spi_stm32_data_ ?
In the end this _dma_channel_data_.dma_tx or _dma_channel_data_.dma_rx are dma stream structure